service_configuration_lib/spark_config.py

```diff
@@ -285,6 +285,17 @@ def _adjust_spark_requested_resources(
         max_cores = int(user_spark_opts.setdefault('spark.cores.max', str(DEFAULT_MAX_CORES)))
         executor_instances = max_cores / executor_cores
     elif cluster_manager == 'kubernetes':
+        # TODO(gcoll|COREML-2697): Consider cleaning this part of the code up
+        # once mesos is not longer around at Yelp.
+        if 'spark.executor.instances' not in user_spark_opts:
+            executor_instances = int(user_spark_opts.get('spark.cores.max', str(DEFAULT_MAX_CORES))) // executor_cores
+            user_spark_opts['spark.executor.instances'] = str(executor_instances)
```

Comment:
Having a Warning message here, would be helpful. This is to build the practice having this way as  preferred way and for consistency.
Indicating that it is recommended to add `spark.executor.instances=xyz` in configs.

Reply:
can we just fix this for users once we're 100% off of mesos?